lib/src/blocs/coins_bloc/coins_bloc.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import '../../constants/string_data.dart';
+import '../../services/coin_service.dart';
+import 'coins_event.dart';
+import 'coins_state.dart';
+
+class CoinsBloc extends Bloc<CoinsEvent, CoinsState> {
+  CoinService service;
+  CoinsBloc({required this.service}) : super(CoinsInitial()) {
+    on<CoinListRequested>((event, emit) async {
+      try {
+        emit(CoinsLoadInProgress());
+        final coins = await service.getCoinsFromAPI(
+            currency: event.currency!,
```

Comment:
why we can be sure that event.currency is not null?

Reply:
i fixed